### PR TITLE
[#IR-333]As a dev, I have added show all retro button for an admin to see all retrospectives

### PR DESCRIPTION
**Pull request by Ravi2714**

### CODE REVIEW

File: apps/retrospective/services/retrospective.go [hunk 1]
```diff
@@ -3,17 +3,19 @@ package services
 import (
 	"encoding/json"
 	"errors"
-	"github.com/jinzhu/gorm"
 	"strconv"
 
+	"github.com/jinzhu/gorm"
+
+	"net/http"
```
Comment (chahat-arora) [inline apps/retrospective/services/retrospective.go]: sort imports properly.

File: apps/retrospective/services/retrospective.go [hunk 2]
```diff
@@ -58,19 +61,25 @@ func (service RetrospectiveService) List(userID uint, perPageString string, page
 		Offset(offset).
 		Select("DISTINCT(retrospectives.*)")
 
-	if isAdmin {
-		err = baseQuery.
-			Find(&retrospectiveList.Retrospectives).
-			Error
+	userTeamQuery := db.Model(&userModels.UserTeam{}).
+		Select("team_id").
+		Where("user_teams.deleted_at IS NULL").
```
Comment (chahat-arora) [inline apps/retrospective/services/retrospective.go]: no need of this.

File: apps/retrospective/services/retrospective.go [hunk 3]
```diff
@@ -58,19 +61,25 @@ func (service RetrospectiveService) List(userID uint, perPageString string, page
...
+		Where("user_id = ? and leaved_at IS NULL", userID).
+		QueryExpr()
 
-	} else {
+	if isAdmin {
 		err = baseQuery.
-			Scopes(retroModels.RetroJoinUserTeams).
-			Where("user_teams.user_id = ?", userID).
+			Where("retrospectives.team_id NOT IN (?)", userTeamQuery).
```
Comment (chahat-arora) [inline apps/retrospective/services/retrospective.go:70]: please, try to optimize this.

File: apps/retrospective/services/retrospective.go [hunk 4]
```diff
@@ -58,19 +61,25 @@ func (service RetrospectiveService) List(userID uint, perPageString string, page
...
 			Preload("Team").
-			Find(&retrospectiveList.Retrospectives).
+			Find(&retrospectiveList.OthersRetrospectives).
 			Error
+
 	}
+	err = baseQuery.
+		Where("retrospectives.team_id IN (?)", userTeamQuery).
```
Comment (chahat-arora) [inline apps/retrospective/services/retrospective.go:80]: same here.

File: apps/retrospective/services/retrospective.go (hunk 3 above)
Comment (chahat-arora) [inline apps/retrospective/services/retrospective.go:70]: handle error also.

**Review by chahat-arora — CHANGES_REQUESTED** [review]

File: apps/retrospective/services/retrospective.go (hunk 1 above)
Comment (Ravi2714) [reply apps/retrospective/services/retrospective.go]: done

File: apps/retrospective/services/retrospective.go (hunk 2 above)
Comment (Ravi2714) [reply apps/retrospective/services/retrospective.go]: removed

File: apps/retrospective/services/retrospective.go (hunk 4 above)
Comment (Ravi2714) [reply apps/retrospective/services/retrospective.go:80]: changed

File: apps/retrospective/services/retrospective.go (hunk 3 above)
Comment (Ravi2714) [reply apps/retrospective/services/retrospective.go:70]: done

**Review by chahat-arora — APPROVED** [review]

**Review by himanshu-garg-165 — APPROVED** [review]

**Pull request merged**